### DISCUSSION TITLE
Loading historical data with docker compose?

### DISCUSSION THREAD

**Discussion by toshioue** [jertel/vuegraf#135] [Q&A] [answered]
Hi,

I was wondering if there was a a command or config to load historical data using docker compose?

**Comment by jertel** [comment]
The docker-compose yaml that comes with this project does not have the command line args specified. You could edit your copy to include them, but you'd need to make sure you remove the args from the docker-compose file after the load completes. Otherwise when you restart it after a reboot, it would reload the history again, causing duplicate data.

Another option is to run the docker-compose as-is (following the instructions in the README.md, and then in a second terminal, run vuegraf again using the `docker run ...` command line (also specified in the README.md) with the load history arg.

**Comment by toshioue** [comment]
I'm not super proficient in docker compose, but what arg would I add to to the docker-compose yaml file to load the data (1st option presented)?

**Comment by jertel** [comment]
Try adding:

```
command: /opt/vuegraf/vuegraf.py /opt/vuegraf/conf/vuegraf.json --historydays 365
```

underneath this line:

```
container_name: vuegraf
```

**Reply by toshioue** [reply]
Thank you,

but I'm see to be getting this error in the logs:

usage: vuegraf.py [-h] [--version] [-v] [-q] [--historydays HISTORYDAYS]
                  [--resetdatabase]
                  configFilename
vuegraf.py: error: unrecognized arguments: /opt/vuegraf/conf/vuegraf.json
usage: vuegraf.py [-h] [--version] [-v] [-q] [--historydays HISTORYDAYS]
                  [--resetdatabase]
                  configFilename
vuegraf.py: error: unrecognized arguments: /opt/vuegraf/conf/vuegraf.json
usage: vuegraf.py [-h] [--version] [-v] [-q] [--historydays HISTORYDAYS]
                  [--resetdatabase]
                  configFilename
vuegraf.py: error: unrecognized arguments: /opt/vuegraf/conf/vuegraf.json
usage: vuegraf.py [-h] [--version] [-v] [-q] [--historydays HISTORYDAYS]
                  [--resetdatabase]
                  configFilename
vuegraf.py: error: unrecognized arguments: /opt/vuegraf/conf/vuegraf.json


----- This is what my docker-compose.yaml file looks like: --------
version: "3.5"
services:
  influxdb:
     image: influxdb
     container_name: influxdb
     ports:
       - 8086:8086
     volumes:
       - influxdb-storage:/var/lib/influxdb2
     environment:
       - INFLUXD_SESSION_LENGTH=432000
     restart: always
  vuegraf:
      image: jertel/vuegraf
      container_name: vuegraf
      command: /opt/vuegraf/vuegraf.py /opt/vuegraf/conf/vuegraf.json --historydays 365
      ports:
       - 8087:8087
      depends_on:
          - influxdb
      volumes:
         - /home/hpo/vuegraf:/opt/vuegraf/conf #This assumes you have created a folder called vuegraf and placed the vu>      restart: always

-------------

**Reply by jertel** [reply]
Try moving the `--historydays 365` in front of the vuegraf.json arg.

**Reply by toshioue** [reply]
My apologies, same error return in the logs :

with command:  /opt/vuegraf/vuegraf.py --historydays 365 /opt/vuegraf/conf/vuegraf.json 

usage: vuegraf.py [-h] [--version] [-v] [-q] [--historydays HISTORYDAYS]
                  [--resetdatabase]
                  configFilename
vuegraf.py: error: unrecognized arguments: /opt/vuegraf/conf/vuegraf.json

**Reply by jertel** [reply, marked as answer]
Sorry, I don't typically use docker-compose. A community member contributed that piece. Try switching to the array syntax like this:

```
      command: 
        - --historydays
        - "365"
        - /opt/data/conf/vuegraf.json
```

I tried it locally and it seemed to accept it.

**Reply by toshioue** [reply]
Thank you so much!

this format worked for me!

**Reply by jertel** [reply]
Thanks for confirming. Don't forget to remove that `command:` section, otherwise when you next run docker-compose again, it will reload history and cause duplicate data.
